src/components/Modal/Modal.vue

```diff
@@ -38,6 +41,16 @@ const props = defineProps({
 
 const emit = defineEmits(['close']);
 
+const modal = ref(null)
+
+onUpdated(() => {
+  if (props.show) {
+    focusModal();
+  }
+});
```

Comment:
`watch` might be a bit cleaner. `onUpdated` is a bit generic

```suggestion
watch(() => props.show, (showValue) => {
  if (showValue) {
    focusModal();
  }
});
```

Reply:
onUpdated was used because it is called after the DOM has finished updating.
I refactored the watch to use nextTick so we can ensure the Modal element is always present in the DOM